### ISSUE TITLE
Results without false positives in the nearest neighbors

### ISSUE THREAD

**Issue by skr998** [wvangansbeke/Unsupervised-Classification#99]
Hello, thanks for your outstanding work.

There is a question about the "Results without false positives in the nearest neighbors" in your paper, because if you remove some samples in the nearest neighbors set, you can't obtain K nearest neighbors for every samples. 

So how did you solve this problem？

**Comment by wvangansbeke** [comment]
Hi @skr998,

If all neighbors are incorrect, we only trained on augmented views for that specific image. Keep in mind, this is only to provide an approximation of the upperbound of our method since this requires lables.

**Comment by skr998** [comment]
OK, you said "If all neighbors are incorrect, we only trained on augmented views for that specific image.", is that means you also only select 1 from the K neighbors set corrected by labels, to compute the scan loss? 

Moreover, you evaluate the model every epoch, and select 5 neighbors to compute the loss together. Have you used the labels to correct this neighbor set in the evaluation phase every epoch? I guess you didn't.

**Comment by wvangansbeke** [comment]
Hi @skr998,

- No, no neighbors are involved in that case. We only used augmentations of the original image.
- No we don't use labels. We simply use the SCAN loss on the validation set to select the best model.
